### PR TITLE
graphite2: Depends on freetype2 when built on Linux

### PR DESCRIPTION
**Pull request by dholm**

### CODE REVIEW

File: Formula/graphite2.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Graphite2 < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "freetype" if OS.linux?
```
Comment (sjackman) [inline Formula/graphite2.rb]: Please use `unless OS.mac?` rather than `if OS.linux?`

**Comment by sjackman** [comment]
What's the build error that you trying to resolve this this PR? Please report the output of ` brew gist-logs` without this PR applied.

**Comment by dholm** [comment]
```
[ 85%] Building C object tests/examples/CMakeFiles/freetype.dir/freetype.c.o                                                                                                                                                                                                                                                  
cd /tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/tests/examples && /home/eholmda/.linuxbrew/Library/Homebrew/shims/linux/super/gcc-5 -DSIZEOF_SIZE_T=8 -I/tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/include -I/tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/src -I/tmp/graphite2-20180828-403
509-186qd7b/graphite2-1.3.12/tests/examples/../../src -I/home/eholmda/.linuxbrew/include/freetype2  -DNDEBUG   -o CMakeFiles/freetype.dir/freetype.c.o   -c /tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/tests/examples/freetype.c
[ 86%] Linking C executable simple                                                                                                                                                                                                                                                                                            
cd /tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/tests/examples && /home/eholmda/.linuxbrew/Cellar/cmake/3.12.1/bin/cmake -E cmake_link_script CMakeFiles/simple.dir/link.txt --verbose=1      
[ 87%] Linking C executable features  
cd /tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/tests/examples && /home/eholmda/.linuxbrew/Cellar/cmake/3.12.1/bin/cmake -E cmake_link_script CMakeFiles/features.dir/link.txt --verbose=1
/home/eholmda/.linuxbrew/Library/Homebrew/shims/linux/super/gcc-5 -DNDEBUG  -rdynamic CMakeFiles/simple.dir/simple.c.o  -o simple -Wl,-rpath,/tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/src ../../src/libgraphite2.so.3.2.1 -lc -lgcc
/home/eholmda/.linuxbrew/Library/Homebrew/shims/linux/super/gcc-5 -DNDEBUG  -rdynamic CMakeFiles/features.dir/features.c.o  -o features -Wl,-rpath,/tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/src ../../src/libgraphite2.so.3.2.1 -lc -lgcc
[ 88%] Linking CXX executable utftest                                               
cd /tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/tests/utftest && /home/eholmda/.linuxbrew/Cellar/cmake/3.12.1/bin/cmake -E cmake_link_script CMakeFiles/utftest.dir/link.txt --verbose=1
/home/eholmda/.linuxbrew/Library/Homebrew/shims/linux/super/g++-5  -DNDEBUG  -rdynamic CMakeFiles/utftest.dir/utftest.cpp.o  -o utftest -Wl,-rpath,/tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/src ../../src/libgraphite2.so.3.2.1 -lc -lgcc
[ 89%] Linking C executable linebreak                                               
cd /tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/tests/examples && /home/eholmda/.linuxbrew/Cellar/cmake/3.12.1/bin/cmake -E cmake_link_script CMakeFiles/linebreak.dir/link.txt --verbose=1
In file included from /tmp/graphite2-20180828-403509-186qd7b/graphite2-1.3.12/tests/examples/freetype.c:5:0:
/usr/include/ft2build.h:56:38: fatal error: freetype/config/ftheader.h: No such file or directory
compilation terminated.                       
make[2]: *** [tests/examples/CMakeFiles/freetype.dir/build.make:66: tests/examples/CMakeFiles/freetype.dir/freetype.c.o] Error 1
```

**Comment by sjackman** [comment]
```
/usr/include/ft2build.h:56:38: fatal error: freetype/config/ftheader.h: No such file or directory
```
Your build is pulling in this header `ft2build.h` from your host system, and that header finds that the header `freetype/config/ftheader.h` is missing. On Ubuntu, both of those files are provided by `libfreetype6-dev`. I'd suggest either reinstalling that package, or removing that package.

`brew linkage graphite2` does not report any dynamic linkage to `freetype2`.

**Comment by dholm** [comment]
@sjackman I don't have root access, that's why I use Linuxbrew. :)

**Comment by iMichka** [comment]
I bet that if freetype is installed somewhere, it will try to link against it. I bet that if you install freetype on Mac and try to build graphite from source you will hit the same problem. 

I am fine to add the dependency here. Upstream has been reluctant to accept these kind of patches so let’s keep this change Linux-only for the moment.

@dholm you need to remove the sha256 line for Linux, and to bump the revision so that a new bottle is built.

Another solution would be to find the flag that explicitly disables building with freetype.

**Comment by dholm** [comment]
@iMichka It appears that only one of the tests use `freetype2`.  There doesn't seem to be a clean way of disabling it either.  There is an option that can be set in CMake, `GRAPHITE2_NFILEFACE`, that will prevent that test from being built but it also disables features in `graphite2` so that doesn't appear to be a good "fix".

**Comment by dholm** [comment]
@iMichka I removed the bottle sha256.

**Comment by sjackman** [comment]
>> brew linkage graphite2 does not report any dynamic linkage to freetype2.

> Another solution would be to find the flag that explicitly disables building with freetype.

>  It appears that only one of the tests use freetype2.

I'd prefer to disable `freetype` if possible, though David reports that it's not straight forward.

Since there's no run-time linkage to `freetype`, it should be a `:build` dependency.

Please describe in a comment above the dependency that it's needed only for a test.

File: Formula/graphite2.rb [hunk 2]
```diff
@@ -11,10 +11,10 @@ class Graphite2 < Formula
     sha256 "bc6822fcaec5128f42706dc99ddf4e7565b4b2dc08d5ecb72757991b3fc94dca" => :high_sierra
     sha256 "7f80d1268cae3d36d362045a08fbb1454cb7249506bce905853204ea401424b0" => :sierra
     sha256 "d6203a6e9563431e67f0a7579e0a94af962c48b0738656459a86277289e9d251" => :el_capitan
-    sha256 "0168ab5d3ef730c9f58e4da0ceaf268c9b81d0df0cf9481a1fd2b7749c21ac9b" => :x86_64_linux
```
Comment (sjackman) [inline Formula/graphite2.rb]: I won't be rebuilding the bottle for this PR, so please revert this chunk.

**Comment by sjackman** [comment]
>> you need to remove the sha256 line for Linux, and to bump the revision so that a new bottle is built.

> @iMichka I removed the bottle sha256.

There's no need to rebuild the bottle for this PR. It does not affect the bottle contents. A squash merge is sufficient.

File: Formula/graphite2.rb (hunk 2 above)
Comment (dholm) [reply Formula/graphite2.rb]: Fixed

**Comment by sjackman** [comment]
Merged. Thanks again for your contribution to Linuxbrew, David! 🏅

**Pull request merged**